Changes needed to properly create the libstdbuf.so shared library on Solaris.

This change has been passed upstream.

--- coreutils-8.16/src/libstdbuf.c.orig	2014-05-08 21:41:23.436563716 -0700
+++ coreutils-8.16/src/libstdbuf.c	2014-05-12 17:44:17.575294603 -0700
@@ -127,7 +127,14 @@
     }
 }
 
-__attribute__ ((constructor)) static void
+#if (__GNUC__ >= 3 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 7) \
+     || 0x590 <= __SUNPRO_C)
+/* Explicitly define to avoid otherwise elided __attribute on SUNPRO_C  */
+# define ATTRIBUTE_CONSTRUCTOR __attribute ((constructor))
+#else
+# error "__attribute ((constructor)) support is required"
+#endif
+static void ATTRIBUTE_CONSTRUCTOR
 stdbuf (void)
 {
   char *e_mode = getenv ("_STDBUF_E");
